File: application/app/Enums/Expenses/ExpenseTableActions.php
<?php

namespace App\Enums\Expenses;

enum ExpenseTableActions: string {
    case Active = 'Set Active';
    case Inactive = 'Set Inactive';
    case Pay = 'Set a Date Paid';
    case Unpay = 'Set a Date Unpaid';
    case Delete = "Delete";
    case Edit = "Edit";
    case EditPayments = "Edit Payments";

    public static function fromName(string $name): ?self
    {
        foreach (self::cases() as $case) {
            if ($case->name === $name) {
                return $case;
            }
        }
        return null;
    }
}

File: server/BuilderServices/ExpenseService/Enums/ExpenseTableActions.cs
using BuilderRepositories;

namespace BuilderServices.ExpenseService.Enums;

public enum ExpenseTableAction
{
    Active,
    Inactive,
    Pay,
    Unpay,
    Delete,
    Edit
}

public static class ExpenseTableActionsHelper
{
    public static string GetActionText(this ExpenseTableAction action, bool recurrenceIsOnce)
    {
        return action switch
        {
            ExpenseTableAction.Active => "Set Active",
            ExpenseTableAction.Inactive => "Set Inactive",
            ExpenseTableAction.Pay => recurrenceIsOnce ? "Mark as Paid" : "Mark a Date Paid",
            ExpenseTableAction.Unpay => recurrenceIsOnce ? "Mark as Unpaid" : "Mark a Date Unpaid",
            ExpenseTableAction.Delete => "Delete",
            ExpenseTableAction.Edit => "Edit",
            _ => throw new GenericException("Invalid expense table action.")
        };
    }
}